### PR TITLE
Fix trailing spaces in call_number fields, remove empty call_number from holdings1_display

### LINKED ISSUE #1505: Call number browse not finding CD- 50000
https://catalog-alma-qa.princeton.edu/browse/call_numbers?search_field=browse_cn&q=CD-+50000 gives no results; this search gives results in catalog.princeton.edu

The record is in the qa catalog: https://catalog-alma-qa.princeton.edu/catalog/99103141233506421

Reported by Brian Newhouse; follow-up requested.

### LINKED ISSUE #1517: Browse related items link displays for empty call_number
@christinach commented on [Thu Mar 28 2019](https://github.com/pulibrary/orangelight/issues/1621)

https://catalog.princeton.edu/catalog/10266460
An empty call_number in this record triggers the 
`Browse related items` link(https://catalog.princeton.edu/browse/call_numbers?q=) 
Also `Where to find it` link(https://catalog.princeton.edu/catalog/10266460/stackmap?loc=sci&cn=) doesn't show the stackmap for Lewis Library.

---

@kevinreiss commented on [Thu Mar 28 2019](https://github.com/pulibrary/orangelight/issues/1621#issuecomment-477788923)

Items without a call number cannot be resolved by the stackmap service. There are a couple of holding location codes [https://github.com/pulibrary/orangelight/blob/2c758d418f386e71498a83a3102fc60bf3b2523c/app/services/stackmap_service.rb#L88] that are resolved by passing the title to the "cn" parameter but other than those items w/o a call number will produce an error.  We should probably add something that fails more gracefully when an item lacks a call number and isn't in a shelve by title location.

---

@tampakis commented on [Fri Mar 29 2019](https://github.com/pulibrary/orangelight/issues/1621#issuecomment-477981287)

I recommend in marc_liberation to modify the code that indexes the `holdings_1display`, `call_number_display`, and `call_number_browse_s` fields to exclude blank strings. If the call number field is nil in a document then the browse related items and where to find it links won't display

### PR DESCRIPTION
**Pull request by christinach**
closes #1505
closes #1517

update fixtures

### CODE REVIEW

**Comment by christinach** [comment]
Regarding issue https://github.com/pulibrary/bibdata/issues/1517
Removing the empty call number fields from holdings1_display, the record page does not include the 'Browse related items' when there is no call_number_browse field.

Screen Shot 2021-07-14 at 5 21 08 PM

Screen Shot 2021-07-14 at 5 20 42 PM

example with mms_id: 99102664603506421

**Comment by christinach** [comment]
I turned this to a draft I want to check one more thing.

**Comment by christinach** [comment]
I opened it for review

File: spec/marc_to_solr/lib/config_spec.rb [hunk 1]
```diff
@@ -170,22 +174,35 @@ def fixture_record(fixture_name)
...
+      expect(@record_call_number2['call_number_display']).to eq(["CD- 50000"])
     end
   end
 
   describe "call_number_browse field" do
-    it "returns the call_number_browse field with k subfield at the end" do
-      record = @indexer.map_record(fixture_record('9957270023506421'))
-      expect(record['call_number_browse_s']).to eq(["6819 Eng 20Q"])
+    it "indexes the call_number_browse field" do
+      expect(@sample40['call_number_browse_s']).to eq([" .B7544 2003q Oversize RA566.27"])
```
Comment (christinach) [inline spec/marc_to_solr/lib/config_spec.rb:189]: @mzelesky for mms_id 9941598513506421 we have :
`<datafield tag="852" ind1="0" ind2=" ">
    <subfield code="b">firestone</subfield>
    <subfield code="c">stacks</subfield>
    <subfield code="h"/>
    <subfield code="i">.B7544 2003q</subfield>
    <subfield code="k">Oversize RA566.27</subfield>
    <subfield code="8">22172120500006421</subfield>
  </datafield>`

h is an empty space in front. Do I keep this empty space and then I would have call_number -> `" .B7544 2003q Oversize RA566.27"` or do I remove the leading spaces as well and then the call_number would be -> `".B7544 2003q Oversize RA566.27"` with no space in front?

Comment (christinach) [reply spec/marc_to_solr/lib/config_spec.rb:189]: In current production we have these spaces. https://catalog.princeton.edu/catalog/4159851/raw
But I think we should't. 
I will update the PR to remove the spaces.

Comment (christinach) [reply spec/marc_to_solr/lib/config_spec.rb]: @mzelesky I dont know if you have time to take a look but I removed all the spaces from empty subfields for the call_number fields. Also for the call number fields in the holdings_1display.

**Review by hackartisan — COMMENTED** [review]
It looks good to me; it would be good to hear from @mzelesky about the leading spaces since we were previously testing that they were present..

**Comment by mzelesky** [comment]
I don't think leading spaces are needed. It makes no sense to me to have a filing space. The record referenced above (https://catalog.princeton.edu/catalog/4159851/raw) is a data error. The info that is supposed to be in subfield h is instead in subfield c.

**Review by mzelesky — APPROVED** [review]
This looks to be a definite improvement over current behavior.

**Pull request merged**